Realm/Realm/DatabaseTypes/InvalidRealmObject.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ////////////////////////////////////////////////////////////////////////////
+// //
+// // Copyright 2021 Realm Inc.
+// //
+// // Licensed under the Apache License, Version 2.0 (the "License")
+// // you may not use this file except in compliance with the License.
+// // You may obtain a copy of the License at
+// //
+// // http://www.apache.org/licenses/LICENSE-2.0
+// //
+// // Unless required by applicable law or agreed to in writing, software
+// // distributed under the License is distributed on an "AS IS" BASIS,
+// // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// // See the License for the specific language governing permissions and
+// // limitations under the License.
+// //
+// ////////////////////////////////////////////////////////////////////////////
+
+namespace Realms
+{
+    /// <summary>
+    /// Special invalid realm object that is used to avoid an exception in WPF
+    /// when deleting an element from a collection bound to UI (<see href="https://github.com/realm/realm-dotnet/issues/1903">#1903</see>).
+    /// </summary>
+    internal class InvalidRealmObject : RealmObjectBase
```

Comment:
The GetHashCode warning is legit - I'm not sure we have good options here. The `GetHashCode` for `RealmObjectBase` will cache the value for the hash code, meaning it'll have some non-zero value. Let's hope there's no UI framework out there that's putting objects in a hashset or we'd be in trouble. For now, what you can do to dismiss the warning is to just return `0` from `GetHashCode`.